api/envoy/config/route/v3/route.proto

```diff
@@ -133,6 +133,13 @@ message RouteConfiguration {
   // :ref:`envoy_v3_api_msg_config.route.v3.VirtualHost`.domains field.
   bool ignore_port_in_host_matching = 14;
 
+  // Normally, virtual host matching is done using the :authority (or
+  // Host: in HTTP < 2) HTTP header. Setting this will instead, use a
+  // different HTTP header for this purpose. This is intended to be
+  // combined with an "early_header_mutation" extension to allow
+  // alternate or simplified host values to be used for host matching.
+  string alternate_header_for_host_matching = 18;
```

Comment:
`vhost_header` is enough. Because most users could get it is header name that will be used for vhost searching from the simple `vhost_header`.

The comments also could be simplified and needn't to refer the early header mutation. The new feature self should be a general/common feature and needn't be bound to early header mutation